Client/Ecosia/UI/Theme/EcosiaThemeManager.swift

```diff
@@ -50,6 +50,11 @@ public final class EcosiaThemeManager: ThemeManager, Notifiable {
         self.userDefaults.register(defaults: [ThemeKeys.systemThemeIsOn: true,
                                               ThemeKeys.NightMode.isOn: NSNumber(value: false)])
 
+        if LegacyThemeManager.instance.systemThemeIsOn {
+            let userInterfaceStyle = UIScreen.main.traitCollection.userInterfaceStyle
+            LegacyThemeManager.instance.current = userInterfaceStyle == .dark ? LegacyDarkTheme() : LegacyNormalTheme()
+        }
+
         changeCurrentTheme(loadInitialThemeType())
```

Comment:
It feels like what you do here is exactly what should happen inside `changeCurrentTheme(loadInitialThemeType())`. I guess this has to do with the cold start, but any reason why not keeping just one of the two?

Reply:
@lucaschifino thanks for asking!
While it may look the same, a small check makes (on our codebase with our references of the `LegacyThemeManager` throughout it) the whole logic behaves unexpectedly (from the User perspective), setting the wrong theme.
Within `loadInitialThemeType()`, a check of a persisted value is made 👇 

`let savedThemeDescription = userDefaults.string(forKey: ThemeKeys.themeName)`.

This check will extract the last saved theme before killing the app, therefore a different one in case we switched at app killed state where no observers were alive.

Reply:
Good input will do that